### PR TITLE
[C-1] Add linked_list to c_dsa

### PR DESCRIPTION
**Pull request by KyryloKupin**
A linked list library to conveniently  deal with linked lists.

### CODE REVIEW

File: linked_list/include/linked_list.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    int size;
+    ListNode* head;
+} ll;
+
+ll *ll_create(int data[], int size);
+void ll_remove_node(ll *ll, ListNode **node);
+void ll_destroy(ll *ll);
+void ll_print(ll* ll);
+
+#endif //LINKED_LIST_H
```
Comment (irog-c) [inline linked_list/include/linked_list.h:19]: Maybe we should put a NL at every EOF.

File: linked_list/src/linked_list.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+    free(ll);
+    ll = NULL;
+}
+
+void
+ll_print(ll* ll)
+{
+    for (ListNode *n = ll->head; n; n = n->next) {
+        printf("%d -> ", n->val);
```
Comment (irog-c) [inline linked_list/src/linked_list.c:52]: What will happen with the last element? If we have [1, 2, 3, 4] then it will print
1 -> 2 -> 3 -> 4 ->
We don't want the last -> so maybe we should handle this? Ofc it is a nitpick 😄

**Comment by irog-c** [comment]
Perhaps we should also add a few more tests? For example we should test if it will still work if we try to remove a NULL node etc.?

**Comment by KyryloKupin** [comment]
> Perhaps we should also add a few more tests? For example we should test if it will still work if we try to remove a NULL node etc.?

Of course, added some more.

File: linked_list/include/linked_list.h (hunk 1 above)
Comment (KyryloKupin) [reply linked_list/include/linked_list.h]: Definitely
